common/menu-entry.c

```diff
@@ -322,42 +322,37 @@ bool menuEntryLoad(menuEntry_s* me, const char* name, bool shortcut) {
 }
 
 void menuEntryParseIcon(menuEntry_s* me) {
-    uint8_t *imageptr = NULL;
-    size_t imagesize = 256*256*3;
-
     if (me->icon_size==0 || me->icon==NULL) return;
 
-    njInit();
+    int w,h,samp;
+    size_t imagesize = 256*256*3;
+    me->icon_gfx = (uint8_t*)malloc(imagesize);
 
-    if (njDecode(me->icon, me->icon_size) != NJ_OK) {
-        njDone();
-        return;
-    }
+    if (me->icon_gfx == NULL) return;
 
-    me->icon_size = 0;
-    free(me->icon);
-    me->icon = NULL;
+    tjhandle _jpegDecompressor = tjInitDecompress();
```

Comment:
Check for failure with nullptr.

Reply:
done

Reply:
do we want to free me-> icon  if _jpegDecompressor fails?

Reply:
same thing if anything else fails before me->icon is freed?

Reply:
No, leave it.

Reply:
k